uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js

```diff
@@ -154,25 +154,29 @@
                 expect(Entitlement.all.calls.argsFor(0)[1]).toEqual(true);
               });
 
-              it('has stored the remainder in each absence type', function () {
+              it('has stored the entitlement, remainder in each absence type which has entitlement', function () {
                 _.forEach(controller.absenceTypes, function (absenceType) {
-                  var remainder = absenceType.remainder;
-
-                  expect(remainder).toBeDefined();
-                  expect(remainder).toEqual(_.find(controller.entitlements, function (entitlement) {
+                  var entitlement = _.find(controller.entitlements, function (entitlement) {
                     return entitlement.type_id === absenceType.id
-                  })['remainder']);
+                  });
+
+                  if (entitlement) {
+                    expect(absenceType.entitlement).toEqual(entitlement['value']);
+                    expect(absenceType.remainder).toEqual(entitlement['remainder']);
+                  }
                 });
               });
 
-              it('has stored the entitlement value in each absence type', function () {
+              it('has stored the 0 value for entitlement, remainder for absence types which does not have entitlement', function () {
                 _.forEach(controller.absenceTypes, function (absenceType) {
-                  var value = absenceType.entitlement;
-
-                  expect(value).toBeDefined();
-                  expect(value).toEqual(_.find(controller.entitlements, function (entitlement) {
+                  var entitlement = _.find(controller.entitlements, function (entitlement) {
```

Comment:
Did not do that because the test case is inside a forEach, so if I create 2 test cases the loop will happen twice. Also beforeEach wont help as the `expect`is inside the loop.